freqtrade/freqai/utils.py

```diff
@@ -191,3 +193,28 @@ def add_feature_trace(fig, fi_df, col):
         fig.update_layout(title_text=f"Best and worst features by importance {pair}")
         label = label.replace('&', '').replace('%', '')  # escape two FreqAI specific characters
         store_plot_file(fig, f"{dk.model_filename}-{label}.html", dk.data_path)
+
+
+def record_params(config: Dict[str, Any], full_path: Path) -> None:
+    """
+    Records run params in the full path for reproducibility
+    """
+    params_record_path = full_path / "run_params.json"
+
+    run_params = {
+        "freqai": config.get('freqai', {}),
+        "timeframe": config.get('timeframe'),
+        "stake_amount": config.get('stake_amount'),
+        "stake_currency": config.get('stake_currency'),
+        "max_open_trades": config.get('max_open_trades'),
+        "pairs": config.get('exchange', {}).get('pair_whitelist')
+    }
+
+    with open(params_record_path, "w") as handle:
+        rapidjson.dump(run_params, handle, indent=4, default=np_encoder,
+                       number_mode=rapidjson.NM_NATIVE)
+
+
+def np_encoder(self, object):
+    if isinstance(object, np.generic):
+        return object.item()
```

Comment:
I don't think the config object will contain numpy objects (if it does, that's clearly not expected and i'd be curious which key that was) ... so i'm not sure why we need that (i think using "str" as default for dump should work as well).